src/main/kotlin/com/demonwav/mcdev/util/McPsiUtil.kt

```diff
@@ -35,26 +35,70 @@ import java.util.stream.Collectors
 @JvmOverloads
 @Contract(value = "null, _ -> null", pure = true)
 fun getClassOfElement(element: PsiElement?, resolveReferences: Boolean = false): PsiClass? {
+    return findParent(element, resolveReferences)
+}
+
+@Contract(value = "null -> null", pure = true)
+fun findReferencedMember(element: PsiElement?): PsiMember? {
+    return findParent(element, true)
+}
+
+@Contract(value = "null -> null", pure = true)
+inline fun <reified T : PsiElement> findParent(element: PsiElement?, resolveReferences: Boolean = false): T? {
     var el = element
     while (el != null) {
         if (resolveReferences && el is PsiReference) {
-            el = el.resolve()
+            el = el.resolve() ?: return null
         }
 
-        if (el is PsiClass) {
+        if (el is T) {
             return el
         }
 
         if (el is PsiFile || el is PsiDirectory) {
             return null
         }
 
-        el = el?.parent
+        el = el.parent
     }
 
     return null
 }
 
+inline fun <reified T : PsiElement> findChild(element: PsiElement): T? {
+    val firstChild = element.firstChild ?: return null
+    return findSibling(firstChild, false)
+}
+
+inline fun <reified T : PsiElement> findSibling(element: PsiElement, strict: Boolean = true): T? {
+    var sibling = if (strict) element.nextSibling else element
+
+    while (sibling != null) {
+        if (sibling is T) {
+            return sibling
+        }
+
+        sibling = sibling.nextSibling
+    }
+
+    return null
+}
+
+inline fun findLastChild(element: PsiElement, condition: (PsiElement) -> Boolean): PsiElement? {
+    var child = element.firstChild
+    var lastChild: PsiElement? = null
+
+    while (child != null) {
+        if (condition.invoke(child)) {
```

Comment:
I don't know if this is really better stylistically, but you can actually treat this just as function (`condition(child)` rather than `condition.invoke(child)`) and I personally like treating it as a function better, as it seems a little more expressive as to what it actually is (or more accurately, what it is intended to do).